sklearn/datasets/openml.py

```diff
@@ -253,7 +253,11 @@ def _download_data_arff(file_id, sparse):
     else:
         return_type = arff.DENSE
 
-    arff_file = arff.load(response, return_type=return_type)
+    if PY2:
+        arff_file = arff.load(response, return_type=return_type)
+    else:
+        arff_file = arff.loads(response.read().decode('utf-8'),
```

Comment:
Sigh. I suppose we need something horrid like `arff.load(codecs.iterdecode(urlopen(url), 'utf-8'))` (which is not officially supported by liac-arff, since it expects a stream, not an iterator over lines, which is what `iterdecode` returns ... but this seems to work).

Remember the goal here is memory efficiency... then we should eventually use caching - and improving liac-arff - to make it runtime efficient.